pkg/goDB/storage/gpfile/gpfile.go

```diff
@@ -50,30 +51,51 @@ type GPFile struct {
 	lastSeekPos int64
 
 	// defaultEncoderType governs how data blocks are (de-)compressed by default
-	defaultEncoderType encoders.Type
-	defaultEncoder     encoder.Encoder
+	defaultEncoderType                 encoders.Type
+	defaultHighCardintalityEncoderType encoders.Type
+	defaultEncoder                     encoder.Encoder
 
 	// accessMode denotes if the file is opened for read or write operations (to avoid
 	// race conditions and unpredictable behavior, only one mode is possible at a time)
 	accessMode int
 }
 
+var defaultHighCardinalityEncoderType = encoders.EncoderTypeNull
```

Comment:
@fako1024 : couldn't resist squeezing it into this commit. This is the way I would approach it at the moment. Still allows us to configure the high-cardinality encoding via a function Option, but doesn't expose it via goProbe's configuration.